cpp/src/DataStorm/TopicFactoryI.cpp

```diff
@@ -30,6 +29,7 @@ TopicFactoryI::createTopicReader(
     shared_ptr<FilterManager> sampleFilterFactories)
 {
     shared_ptr<TopicReaderI> reader;
+    auto instance = getInstance();
```

Comment:
Not directly from this PR.

I am not a fan of this getInstance function. It would be more readable to expand it inline, as in:
```cpp
auto instance = _instance.lock();
assert(instance); // comment explaining why it can't be null here
```